crates/driver/src/infra/api/routes/settle/mod.rs

```diff
@@ -8,41 +8,93 @@ use {
             observe,
         },
     },
+    tokio::sync::{mpsc, oneshot},
     tracing::Instrument,
 };
 
 pub(in crate::infra::api) fn settle(router: axum::Router<State>) -> axum::Router<State> {
     router.route("/settle", axum::routing::post(route))
 }
 
+pub(in crate::infra::api) struct QueuedSettleRequest {
+    state: State,
+    req: dto::Solution,
+    response_sender: oneshot::Sender<Result<(), competition::Error>>,
+}
+
+pub(in crate::infra::api) fn create_settle_queue_sender(
```

Comment:
The settle queue seems like something that should live inside the `Solver` struct and should not be defined in the code for the REST API handler as it's domain specific code.

Reply:
Makes sense. But probably not the `Solver`, but `Competition`, since the settlement queue task requires access to the `Competition::settle` function and the `Ethereum` instance, which is part of this struct, including the `Solver` itself and the settlements vector.